Python/Product/Analysis/LanguageServer/Server.cs

```diff
@@ -1275,6 +1260,57 @@ string prefix
             }
         }
 
+        private void BuildMarkdownSignature(SignatureHelp signatureHelp) {
+            foreach (var s in signatureHelp.signatures) {
+                // Recostruct full signature so editor can display current parameter
+                var sb = new StringBuilder();
+
+                if (s.documentation != null) {
+                    s.documentation.value = _textConverter.ToMarkdown(s.documentation.value);
+                }
+                sb.Append(s.label);
+                sb.Append('(');
+                if (s.parameters != null) {
+                    foreach (var p in s.parameters) {
+                        if (sb[sb.Length - 1] != '(') {
+                            sb.Append(", ");
+                        }
+                        sb.Append(p.label);
+                        if (p.documentation != null) {
+                            p.documentation.value = _textConverter.ToMarkdown(p.documentation.value);
+                        }
+                    }
+                }
+                sb.Append(')');
+                s.label = sb.ToString();
+            }
+        }
+
+        private void IfTestWaitForAnalysisComplete() {
+            if (_testEnvironment) {
+                WaitForDirectoryScanAsync().Wait();
+                WaitForCompleteAnalysisAsync().Wait();
+            }
+        }
+
+        private PythonAst GetParseTree(IPythonProjectEntry entry, Uri documentUri, out int? version) {
+            version = null;
+            PythonAst tree = null;
+            var parse = entry.WaitForCurrentParse(_clientCaps?.python?.completionsTimeout ?? Timeout.Infinite, CancellationToken);
```

Comment:
Completions may not be the only timeout we want here (e.g. we have a longer timeout for hover in VS today). This may need to be a parameter